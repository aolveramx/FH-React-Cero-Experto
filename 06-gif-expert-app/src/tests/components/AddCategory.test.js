import '@testing-library/jest-dom'
import { shallow } from "enzyme";
import AddCategory from "../../components/AddCategory";

describe('Tests in <AddCategory />', () => {
  
  const setCategories = jest.fn()
  let wrapper = shallow(<AddCategory setCategories={setCategories} />)

  beforeEach(() => {
    jest.clearAllMocks()
    wrapper = shallow(<AddCategory setCategories={setCategories} />)
  })

  test('should render successful', () => {

    expect(wrapper).toMatchSnapshot()

  });

  test('should get input from textarea', () => {
    
    const input = wrapper.find('input')
    const value = 'Hola Value'
    input.simulate('change', {target: {value}})

    expect(wrapper.find('p').text().trim()).toBe(value)

  });

  test('should not post submit info', () => {
    
    wrapper.find('form').simulate('submit', { preventDefault(){} })

    expect(setCategories).not.toHaveBeenCalled()

  });

  test('should call setCategories and empty textarea', () => {

    const value = 'Hola Value'

    // 1. simular el inputChange
    wrapper.find('input').simulate('change', {target: {value}})

    // 2. simular el submit
    wrapper.find('form').simulate('submit', { preventDefault(){} })

    // 3. setCategories se debe de haber llamado
    expect(setCategories).toHaveBeenCalled()
    expect(setCategories).toHaveBeenCalledTimes(1)
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function))

    // 4. El valor del input debe estar
    expect(wrapper.find('input').prop('value')).toBe('')

  });

}); 