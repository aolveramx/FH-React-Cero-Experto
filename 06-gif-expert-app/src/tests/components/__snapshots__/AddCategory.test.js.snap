// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <AddCategory /> should render successful 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
